hs_core/management/commands/ingest_bag.py

```diff
@@ -0,0 +1,103 @@
+"""
+This command ingests a bag into a resource.  You can provide a resource_id to ingest the bag or the command will create
+a new resource to ingest the bag into.
+"""
+import os
+
+from django.core.exceptions import ValidationError
+from django.core.files.uploadedfile import UploadedFile
+from django.core.management.base import BaseCommand
+
+from hs_core.hydroshare import delete_resource_file, User, add_file_to_resource, create_resource
+from hs_composite_resource.models import CompositeResource
+from zipfile import ZipFile
+from pathlib import Path
+
+from hs_core.views.utils import ingest_bag
+
+
+def extract_resource_id(bag_file_path):
+    """Extracts the resource id by reading the root directory name inside the zip"""
+    with ZipFile(bag_file_path, 'r') as zip_file:
+        resource_id = Path(zip_file.namelist()[0]).parts[0]
+    return resource_id
+
+
+def is_hydrohsare_bagit(bag_file_path):
+    """Checks for a hydroshare bagit file structure in a zip"""
+    resource_id = extract_resource_id(bag_file_path)
```

Comment:
We use resource id as the bag file name, so I think you can directly extract resource id from the bag file name without need to extract it from inside the zip file.

Reply:
I understand that assumption may restrict the general applicability of this bag ingestion code, but for bags created in HydroShare, the resource id is used as the bag name, so it is probably a good thing to add this as the default option and potentially provide a flag like extract_resource_id to allow support of ingestion of bags created outside HS that don't follow HS's bag naming convention.

Reply:
I feel the name of the zip file is too risky to assume it's the resource_id.  If you download a bag twice, it'll be renamed in your downloads directory.  Anything could happen.  It's not a big deal to peek inside and get a more reliable resource_id.  I'm checking inside of it anyways to ensure it follows the bagit structure.